### PR TITLE
Allow user to pass in package URL for package reading

### PR DESCRIPTION
**Pull request by LitoMore**
Related to #176 

### Changes

- Add `importMeta` option for allowing users to pass in their package URL
- Apply `importMeta` option to all the test code

### Spec

- `importMeta` is required
- Its value should be `import.meta`

### CODE REVIEW

**Comment by LitoMore** [comment]
We already have a `pkg` option for users to pass in their package object, what about let users read the package at their side, then pass it to our package?

File: index.js [hunk 1]
```diff
@@ -97,18 +99,20 @@ const validateFlags = (flags, options) => {
...
 
-	const foundPackage = readPackageUpSync({
+	const foundPackage = readPackageSync({
+		cwd: options.packagePath ? dirname(fileURLToPath(options.packagePath)) : process.cwd(),
 		normalize: false
 	});
 
 	options = {
-		pkg: foundPackage ? foundPackage.packageJson : {},
+		packagePath: process.cwd(),
```
Comment (LitoMore) [inline index.js:114]: The default value is `process.cwd()`, but it's useless. Do we need to set it as a required option?

**Comment by sindresorhus** [comment]
I think we should accept `import.meta`, not `import.meta.url`. `import.meta` might get useful properties we might want to use in the future. It's more future proof this way. We can just call the option `importMeta` and make it required.

File: index.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+import {fileURLToPath} from 'node:url';
 import buildParserOptions from 'minimist-options';
 import parseArguments from 'yargs-parser';
 import camelCaseKeys from 'camelcase-keys';
 import decamelize from 'decamelize';
 import decamelizeKeys from 'decamelize-keys';
 import trimNewlines from 'trim-newlines';
 import redent from 'redent';
-import {readPackageUpSync} from 'read-pkg-up';
+import {readPackageSync} from 'read-pkg';
```
Comment (sindresorhus) [inline index.js:10]: The `cli.js` is not guaranteed to be in the same directory as package.json.

File: index.d.ts [hunk 3]
```diff
@@ -20,13 +20,21 @@ export interface Flag<Type extends FlagType, Default> {
...
 type NumberFlag = Flag<'number', number>;
 type AnyFlag = StringFlag | BooleanFlag | NumberFlag;
 type AnyFlags = Record<string, AnyFlag>;
 
 export interface Options<Flags extends AnyFlags> {
+	/**
+	Directory to start looking for the module package.json file.
+
+	This option is required, its value must be `import.meta`.
+	*/
```
Comment (sindresorhus) [inline index.d.ts:35]: Docs are not correct.

File: index.d.ts [hunk 4]
```diff
@@ -20,13 +20,21 @@ export interface Flag<Type extends FlagType, Default> {
 	readonly isMultiple?: boolean;
 }
 
+type ImportMeta = {url: string};
```
Comment (sindresorhus) [inline index.d.ts:23]:
```suggestion
type ImportMeta = typeof import.meta;
```

?

Comment (LitoMore) [reply index.d.ts]: It looks like it's not correct grammar.

Comment (LitoMore) [reply index.d.ts]: How about use `Record<string, unknown> & {url: string}` instead?

Comment (LitoMore) [reply index.d.ts]: We could use `ImportMeta` directly.

The type definitions is here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/globals.d.ts#L29

File: index.js [hunk 5]
```diff
@@ -97,13 +99,18 @@ const validateFlags = (flags, options) => {
...
 
-const meow = (helpText, options) => {
+const meow = (helpText, options = {}) => {
 	if (typeof helpText !== 'string') {
 		options = helpText;
 		helpText = '';
 	}
 
+	if (!(options.importMeta && options.importMeta.url)) {
+		throw new TypeError('`importMeta` options is required, its value must be `import.meta`');
```
Comment (sindresorhus) [inline index.js:109]:
```suggestion
		throw new TypeError('The `importMeta` option is required. Its value must be `import.meta`.');
```

File: index.d.ts [hunk 6]
```diff
@@ -27,6 +27,13 @@ type AnyFlag = StringFlag | BooleanFlag | NumberFlag;
 type AnyFlags = Record<string, AnyFlag>;
 
 export interface Options<Flags extends AnyFlags> {
+	/**
+	Your `import.meta` used for start looking for the module package.json file.
+
+	Set it to `import.meta` to find the module package.json file.
```
Comment (sindresorhus) [inline index.d.ts:33]: These two sentence say almost the same thing.

File: readme.md [hunk 7]
```diff
@@ -126,6 +86,14 @@ Shortcut for the `help` option.
 
 Type: `object`
 
+##### importMeta
+
+Type: `object`
+
+Your `import.meta` used for start looking for the module package.json file.
+
+This option is required, its value must be `import.meta`.
```
Comment (sindresorhus) [inline readme.md:95]: Readme and index.d.ts should be in sync.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts [hunk 8]
```diff
@@ -27,6 +27,13 @@ type AnyFlag = StringFlag | BooleanFlag | NumberFlag;
 type AnyFlags = Record<string, AnyFlag>;
 
 export interface Options<Flags extends AnyFlags> {
+	/**
+	Your `import.meta` used for start looking for the module package.json file.
+
+	This option is required, its value must be `import.meta`.
```
Comment (sindresorhus) [inline index.d.ts:33]:
```suggestion
	Pass in `import.meta`. This is used to find the correct package.json file.
```

**Pull request merged**